pgpainless-core/src/main/java/org/pgpainless/signature/consumer/SignatureValidator.java

```diff
@@ -150,16 +150,18 @@ public void verify(PGPSignature signature) throws SignatureValidationException {
      * Verify that a signature has an acceptable structure.
      *
      * @param signingKey signing key
-     * @param policy policy
+     * @param policy     policy
      * @return validator
      */
     public static SignatureValidator signatureStructureIsAcceptable(PGPPublicKey signingKey, Policy policy) {
         return new SignatureValidator() {
             @Override
             public void verify(PGPSignature signature) throws SignatureValidationException {
                 signatureIsNotMalformed(signingKey).verify(signature);
-                signatureDoesNotHaveCriticalUnknownNotations(policy.getNotationRegistry()).verify(signature);
-                signatureDoesNotHaveCriticalUnknownSubpackets().verify(signature);
+                if (signature.getVersion() == 4) {
```

Comment:
Can you change this to `if (signature.getVersion() >= 4) {` please, so that the requirements are still checked in version 5+ signatures once they become a thing?

Reply:
I was thinking of doing this, but we don't really know what version 5 signatures are going to look like. Perhaps they will be completely different from v4 and won't contain subpackets?

Reply:
Yeah I understand. Still, I believe we better accidentally reject version 5 signatures when we encounter them, then to erroneously accept malformed v5 sigs. Once v5 is finalized, we can still fix the rules down anyways.